programs/cleos/httpc.cpp

```diff
@@ -123,7 +123,7 @@ namespace eosio { namespace client { namespace http {
    resolved_url resolve_url( const http_context& context, const parsed_url& url ) {
       tcp::resolver resolver(context->ios);
       boost::system::error_code ec;
-      auto result = resolver.resolve(url.server, url.port, ec);
+      auto result = resolver.resolve(tcp::v4(), url.server, url.port, ec);
```

Comment:
The problem here was that if the resolver returns IPv6 address `keosd` binds to IPv6 address and then `cleos` cannot connect to `keosd` because it uses IPv4. It happened only on my computer, because somehow default address for `localhost` was IPv6-address, it's perfectly legit but not typical. So, I just restricted the resolver to always return IPv4 (actually Bart suggested it).
More fancy solution would be to make `cleos` to use IPv6 if `localhost` is defaulted to IPv6-address. But I don't think that in this case it makes any practical difference. So, this fix is simple, but it solved my problem.